### DISCUSSION TITLE
Missing dataType param

### DISCUSSION THREAD

**Discussion by razvn** [huntabyte/shadcn-svelte#1552] [Help]
### Change Type

Addition

### Proposed Changes

I tried this exemple and got an error... with Svelte 5.
Turns out it needs: `dataType: 'json'` to work:
```
	const form = superForm(data, {
		dataType: 'json',
		validators: zodClient(formSchema),
	});
```

**Comment by huntabyte** [comment]
Which example did you try?

**Comment by razvn** [comment]
Oh it didn’t link to the documentation line ?
https://github.com/huntabyte/shadcn-svelte/blob/5cd7061ec8025c7fd75e5ce9eeae232191ebde5a/sites/docs/src/content/components/form.md?plain=1#L121

The from comportement exemple.

**Comment by ieedan** [comment]
@razvn I think the problem is that you haven't read the example the whole way though. If you paste the client side code from the example straight into a page it won't work because it is meant to be a component.

To fix this you should use that code in a component as intended or adapt it by changing the code like so:
```diff
<script lang="ts">
	import * as Form from '$lib/components/ui/form/index.js';
	import { Input } from '$lib/components/ui/input/index.js';
	import { formSchema, type FormSchema } from './schema';
	import { superForm } from 'sveltekit-superforms';
	import { zodClient } from 'sveltekit-superforms/adapters';

+	let { data } = $props();
-   export let data: SuperValidated<Infer<FormSchema>>;

+	const form = superForm(data.form, {
-   const form = superForm(data, {
		validators: zodClient(formSchema)
	});

	const { form: formData, enhance } = form;
</script>

<form method="POST" use:enhance>
	<Form.Field {form} name="username">
		<Form.Control>
			{#snippet children({ props })}
				<Form.Label>Username</Form.Label>
				<Input {...props} bind:value={$formData.username} />
			{/snippet}
		</Form.Control>
		<Form.Description>This is your public display name.</Form.Description>
		<Form.FieldErrors />
	</Form.Field>
	<Form.Button>Submit</Form.Button>
</form>
```
